cdap-api/src/main/java/io/cdap/cdap/api/security/credential/NotFoundException.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.security.credential;
+
+/**
+ * An exception which is thrown when a resource is not found.
+ */
+public class NotFoundException extends Exception {
```

Comment:
are we implementing a NotFoundException here instead of reusing io.cdap.cdap.common.NotFoundException because we don't want to add a dependency on cdap-common?

I think the main advantage of io.cdap.cdap.common.NotFoundException is that it implements HttpErrorStatusProvider, so when it's thrown from an http handler, the caller gets a 404. If io.cdap.cdap.api.security.credential.NotFoundException I think the caller would get a 500

Reply:
We cannot add `cdap-common` as a dependency because `cdap-api` is higher up in the dependency tree and would cause circular dependencies. In this case, the expected behavior will be to catch this `NotFoundException` and propagate the `io.cdap.cdap.common.NotFoundException` in its place at the handler level.